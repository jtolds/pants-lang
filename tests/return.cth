myfunc = { |x, y| function {
  print "x" x
  if (< x 10) {
    return y
  }
  print "y"
}}

print "return" myfunc(1,2)
print.
print "return" myfunc(10,2)
