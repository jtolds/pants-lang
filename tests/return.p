# PANTS OPTIONS: --skip-prelude

function = {|func|
  {|:(largs); :(rargs), ::(kwargs)| func(:(largs); cont, :(rargs), ::(kwargs))}
}

myfunc = function { |return, x, y|
  println "x" x
  if (< x 10) {
    return y
  }
  println "y"
}

println "return" myfunc(1,2)
println.
println "return" myfunc(10,2)
