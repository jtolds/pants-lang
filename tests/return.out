x 1
return 2

x 10
y
return y

return code: 0
